distribution/build.gradle

```diff
@@ -283,6 +283,9 @@ configure(subprojects.findAll { ['archives', 'packages'].contains(it.name) }) {
           if (it.relativePath.segments[-2] == 'bin') {
             // bin files, wherever they are within modules (eg platform specific) should be executable
             it.mode = 0755
+          } else if (platform == 'darwin' && it.relativePath.segments[-2] == 'MacOS') {
```

Comment:
I wonder if this should be combined with previous `if` (`||`) so that we don't ever have to worry about maintenance on the previous block being forgotten in this block too? I can't think of a case where they'd deviate, but we can separate them if they need to do that.